Lbf fixup

lgsmith opened the pull request:
---
name: lbf fixup
about: fixes subsetting in prebuilt tool `long-bond-finder`
title: 'Long bond finder subset fix'
labels: 'bug-fix'
assignees: ''
---
# Found a small but critical error in long-bond finder that makes its subsetting selection fail--it didn't call prune-bonds after subsetting.

## Changes proposed in this pull request
    - added call to prune bonds for scope if scope has bonds.
        - If model doesn't have all bonds, this test is skipped.


## Checklist
    - no open issue is addressed.
    - Have you applied a formatter to your code (e.g. flake8 or black for python, clang-format for C++)? Yes.

lgsmith committed:
Made changes to how selection is analyzed for connectivity. Changed
messaging output.

lgsmith committed:
Completed typo check.

lgsmith committed:
Merge branch 'main' into lbf-fixup

lgsmith committed:
Fixed heinous bug that prevented subsetting from working correctly.

agrossfield reviewed (commented):
It looks fine, but I have a very minor request:

Around lines 239 and 241, you've got if test without braces to indicate scope. It's correct as is, but please put the {} in, so someone in the future doesn't make a silly mistake.

Also, most of the changes are braces moving from the same line to the next line. I don't care one way or the other, but I thought it was weird.

Cheers,

Alan

lgsmith committed:
Changed braces so they look a bit less crazy.

lgsmith commented:
I think my formatter did that. It seems to prefer:
```
if (test)
{
   statement;
}
```
to 
```
if (test) {
  statement;
}
```
I changed them all to the latter, and added braces to each clause. Is that OK?
